app/models/settings/smtp.rb

```diff
@@ -0,0 +1,18 @@
+module Settings
+  class SMTP < RailsSettings::Base
+    self.table_name = :settings_smtps
+
+    OPTIONS = %i[address port authentication user_name password domain].freeze
+
+    # The configuration is cached, change this if you want to force update
+    # the cache, or call Settings::Smtp.clear_cache
+    cache_prefix { "v1" }
+
+    field :address, type: :string, default: ApplicationConfig["SMTP_ADDRESS"]
+    field :port, type: :integer, default: ApplicationConfig["SMTP_PORT"]
+    field :authentication, type: :string, default: ApplicationConfig["SMTP_AUTHENTICATION"]
+    field :user_name, type: :string, default: ApplicationConfig["SMTP_USER_NAME"]
+    field :password, type: :string, default: ApplicationConfig["SMTP_PASSWORD"]
```

Comment:
I wold add validation to make sure both of these are specified if either is

Reply:
So I added it and it workes in theory in the console but it wasn't exactly working via the UI. The UI just wouldn't let me empty a field and I believe it's happening to all fields. I would like to ship this so the folks can do some FOSS test and I can address this and that existing problem in a different PR.